### PR TITLE
Support up to 5 RRs for XL compute and above

### PR DESCRIPTION
**Pull request by joshenlim**
As per PR title
- If on small compute (and medium, large), can deploy up to 2 RRs
- From XL compute and above, can deploy up to 5 RRs
- The necessary validations for downgrading compute should hold in place

Updated message when trying to deploy a third replica while on Small compute:
Screenshot 2024-09-12 at 16 51 30

Updated message when trying to downgrade from XL while having more than 2 replicas
Screenshot 2024-09-12 at 17 09 53

Will use the same message when trying to downgrade to micro from XL while having more than 2 replicas
image

### CODE REVIEW

File: apps/studio/components/interfaces/Settings/Addons/ComputeInstanceSidePanel.tsx [hunk 1]
```diff
@@ -163,12 +163,18 @@ const ComputeInstanceSidePanel = () => {
   const selectedCompute = availableOptions.find((option) => option.identifier === selectedOption)
   const hasChanges =
     selectedOption !== (subscriptionCompute?.variant.identifier ?? defaultInstanceSize)
-  const hasReadReplicas = (databases ?? []).length > 1
-
-  const blockDowngradeDueToPitr =
-    pitrAddon !== undefined && ['ci_micro', 'ci_nano'].includes(selectedOption) && hasChanges
+  const numReadReplicas = 3 // (databases ?? []).filter((db) => db.identifier !== projectRef).length
```
Comment (ivasilov) [inline apps/studio/components/interfaces/Settings/Addons/ComputeInstanceSidePanel.tsx:166]: Is this supposed to be hardcoded?

Comment (joshenlim) [reply apps/studio/components/interfaces/Settings/Addons/ComputeInstanceSidePanel.tsx:166]: god dang it sorry!

Comment (joshenlim) [reply apps/studio/components/interfaces/Settings/Addons/ComputeInstanceSidePanel.tsx]: fixed now haha 😅

**Review by alaister — APPROVED** [review]
LGTM!

- Tested all cases in PR description

**Pull request merged**